be/src/formats/orc/apache-orc/c++/src/Compression.cc

```diff
@@ -475,37 +493,53 @@ bool DecompressionStream::Skip(int count) {
     return true;
 }
 
-/** There are three possible scenarios when seeking a position:
-   * 1. The seeked position is already read and decompressed into
-   *    the output stream.
-   * 2. It is already read from the input stream, but has not been
+/** There are four possible scenarios when seeking a position:
+   * 1. The chunk of the seeked position is the current chunk that has been read and
+   *    decompressed. For uncompressed chunk, it could be partially read. So there are two
+   *    sub-cases:
+   *    a. The seeked position is inside the uncompressed buffer.
+   *    b. The seeked position is outside the uncompressed buffer.
+   * 2. The chunk of the seeked position is read from the input stream, but has not been
    *    decompressed yet, ie. it's not in the output stream.
-   * 3. It is not read yet from the inputstream.
+   * 3. The chunk of the seeked position is not read yet from the input stream.
    */
 void DecompressionStream::seek(PositionProvider& position) {
-    size_t seekedPosition = position.current();
-    // Case 1: the seeked position is the one that is currently buffered and
-    // decompressed. Here we only need to set the output buffer's pointer to the
-    // seeked position. Note that after the headerPosition comes the 3 bytes of
+    size_t seekedHeaderPosition = position.current();
+    // Case 1: the seeked position is in the current chunk and it's buffered and
+    // decompressed/uncompressed. Note that after the headerPosition comes the 3 bytes of
     // the header.
-    if (headerPosition == seekedPosition && inputBufferStartPosition <= headerPosition + 3 && inputBufferStart) {
-        position.next();                     // Skip the input level position.
+    if (headerPosition == seekedHeaderPosition && inputBufferStartPosition <= headerPosition + 3 && inputBufferStart) {
```

Comment:
it would be better to use a meaningful const symbol instead of a bare 3. hard to understand.

Reply:
It's the code merged from apache/orc. Better to be consisten with it for later merge.